tests/unittests/test_net.py

```diff
@@ -1905,18 +1923,28 @@ def test_all_config(self, caplog):
             not in caplog.text
         )
 
-    def test_v4_and_v6_static_config(self, caplog):
-        entry = NETWORK_CONFIGS["v4_and_v6_static"]
+    @pytest.mark.parametrize(
+        "yaml_file,network_config",
+        [
+            ("yaml_v1", "v1_ipv4_and_ipv6_static"),
+            ("yaml_v2", "v2_ipv4_and_ipv6_static"),
+        ],
+    )
+    def test_ipv4_and_ipv6_static_config(
+        self, yaml_file, network_config, caplog
+    ):
+        entry = NETWORK_CONFIGS[network_config]
         found = self._render_and_read(
-            network_config=yaml.safe_load(entry["yaml"])
+            network_config=yaml.safe_load(entry[yaml_file])
         )
         self._compare_files_to_expected(entry[self.expected_name], found)
         self._assert_headers(found)
         expected_msg = (
             "Network config: ignoring iface0 device-level mtu:8999"
             " because ipv4 subnet-level mtu:9000 provided."
         )
-        assert expected_msg in caplog.text
+        if yaml_file == "yaml_v1":
+            assert expected_msg in caplog.text
```

Comment:
No test for `yaml_v2`?

Reply:
We get no warning in v2 because we wind up testing slightly different things.  v2 has no concept of interface level and subnet level MTU, so that's not something we can test for in v2.

yaml v1:
```
            version: 1
            config:
              - type: 'physical'
                name: 'iface0'
                mtu: 8999
                subnets:
                  - type: static
                    address: 192.168.14.2/24
                    mtu: 9000
                  - type: static
                    address: 2001:1::1/64
                    mtu: 1500
```

yaml v2:
```
            version: 2
            ethernets:
                iface0:
                    addresses:
                    - 192.168.14.2/24
                    - 2001:1::1/64
                    mtu: 9000
```